### PR TITLE
[3006.x] Remove deprecated pyvenv and add venv support

### LINKED ISSUE #51742: Py3 should prefer venv module to virtualenv binary
In Python 3 the `virtualenv` script was replaced by `pyvenv`, and then in 3.6 deprecated in favour of the builtin `venv` module.

The latter should be preferred, but at minimum `pyvenv` should be added to `virtualenv_mod.KNOWN_BINARY_NAMES`.

### LINKED ISSUE #66132: [BUG] virtualenv state does not work
**Description**
virtualenv state does not run, because virtualenv module is not available. When I run the state I get
```
     Comment: State 'virtualenv.managed' was not found in SLS 'btl.install_python_libs'
                                               Reason: 'virtualenv.managed' is not available.          
```
Does it have anything to do with installed modules?

**Setup**
So far I have only seen this on Windows, not sure if I tested the same version on Linux.
Installer downloaded from https://docs.saltproject.io/salt/install-guide/en/latest/topics/downloads.html
Installer is https://repo.saltproject.io/salt/py3/windows/latest/Salt-Minion-3006.7-Py3-AMD64-Setup.exe
Windows minions and Linux master are running 3006.7 (Sulphur).

**Steps to Reproduce the behavior**
Run a state with virtualenv.managed.
Result: get an error

**Expected behavior**
Should install/create/update a virtualenv.

**Versions Report**
This is from salt-call on the minion. The issue is that the minion is lacking the virtualenv module.

```yaml
Salt Version:                                                                                                                     Salt: 3006.7                                                                                                                                                                                                                          Python Version:                                                                                                                 Python: 3.10.13 (heads/main:3079fcb, Feb 19 2024, 03:36:00) [MSC v.1937 64 bit (AMD64)]                                                                                                                                                 Dependency Versions:                                                                                                              cffi: 1.14.6                                                                                                        cherrypy: 18.6.1                                                                                                        dateutil: 2.8.1                                                                                                        docker-py: Not Installed                                                                                                    gitdb: 4.0.7                                                                                                        gitpython: 3.1.41                                                                                                          Jinja2: 3.1.3                                                                                                          libgit2: Not Installed                                                                                             looseversion: 1.0.2                                                                                                         M2Crypto: Not Installed                                                                                                     Mako: Not Installed                                                                                                  msgpack: 1.0.2                                                                                                     msgpack-pure: Not Installed                                                                                             mysql-python: Not Installed                                                                                                packaging: 22.0                                                                                                         pycparser: 2.21                                                                                                          pycrypto: Not Installed                                                                                             pycryptodome: 3.19.1                                                                                                          pygit2: Not Installed                                                                                             python-gnupg: 0.4.8                                                                                                           PyYAML: 6.0.1                                                                                                            PyZMQ: 25.0.2                                                                                                          relenv: 0.15.1                                                                                                           smmap: 4.0.0                                                                                                          timelib: 0.2.4                                                                                                          Tornado: 4.5.3                                                                                                              ZMQ: 4.3.4                                                                                                                                                                                                                           System Versions:                                                                                                                  dist: msys_nt-10.0-19042 3.1.7                                                                                        locale: utf-8                                                                                                          machine: AMD64                                                                                                          release: 10                                                                                                              system: Windows                                                                                                        version: 10 10.0.19042 SP0 Multiprocessor Free      
```
</details>

Edit: I see the virtualenv_mod is available in the code https://github.com/saltstack/salt/blob/0d518807e2c246856d44e247c627d3aa8b04ca8c/salt/modules/virtualenv_mod.py#L506
but somehow it is missing from the package?

### PR DESCRIPTION
**Pull request by Akm0d**
### What issues does this PR fix or reference?
Fixes #66132
Fixes #51742

### Previous Behavior
The virtualenv module used the deprecated pyvenv library.

### New Behavior
The virtualenv module uses the builtin venv library.

### Merge requirements satisfied?
**[NOTICE] Bug fixes or features added to Salt require tests.**

- [x] Docs
- [x] Changelog - https://docs.saltproject.io/en/master/topics/development/changelog.html
- [x] Tests written/updated

### Commits signed with GPG?
Yes

Please review [Salt's Contributing Guide](https://docs.saltproject.io/en/master/topics/development/contributing.html) for best practices, including the
[PR Guidelines](https://docs.saltproject.io/en/master/topics/development/pull_requests.html).

See GitHub's [page on GPG signing](https://help.github.com/articles/signing-commits-using-gpg/) for more information about signing commits with GPG.

### CODE REVIEW

**Review by dmurphy18 — DISMISSED** [review]

File: salt/modules/virtualenv_mod.py [hunk 1]
```diff
@@ -82,8 +82,6 @@ def create(
     never_download=None,
     prompt=None,
     pip=False,
-    symlinks=None,
-    upgrade=None,
```
Comment (twangboy) [inline salt/modules/virtualenv_mod.py:86]: I think these need to go through a deprecation process. You can't just remove parameters.

**Review by twangboy — CHANGES_REQUESTED** [review]

File: tests/unit/modules/test_virtualenv_mod.py [hunk 2]
```diff
@@ -17,6 +17,8 @@
 from tests.support.mock import MagicMock, patch
 from tests.support.unit import TestCase
 
+VENV_BIN = f"{sys.executable} -m venv"
```
Comment (twangboy) [inline tests/unit/modules/test_virtualenv_mod.py:20]: The module is modified to use virtualenv, yet we're using venv for the tests. Is that right?

**Review by twangboy — CHANGES_REQUESTED** [review]

**Review by dmurphy18 — COMMENTED** [review]
Looks good, wonder if we have sufficient test coverage with existing tests

**Review by twangboy — APPROVED** [review]

**Review by dmurphy18 — APPROVED** [review]

**Comment by OrangeDog** [comment]
The documentation is no longer correct. It says the default is to use `virtualenv`, but it now is not if not installed.

It also appears that there's no way to use the `venv` module of a different Python (e.g. the system Python)?

People will have to workaround by creating custom `pyvenv` wrappers.

**Pull request merged**